### ISSUE TITLE
Can't get data from LiteDB collection nested in an object

### ISSUE THREAD

**Issue by quasar-pankaj** [mbdavid/LiteDB#1118]
[LiteDBDemo.zip](https://github.com/mbdavid/LiteDB/files/2548206/LiteDBDemo.zip)
Hello Mauricio,
Let me first say that litedb is a very good concept. it requires no configuration and no headaches associated with relational databases. Most of all it is super simple to use. I am very new to the concept of nosql databases and wrote a small console application to test drive litedb. This is documented on stack overflow at the following url:- https://stackoverflow.com/questions/53131085/cant-get-data-from-litedb-collection-nested-in-an-object . I was expecting that I had missed something so there would be some guidance in this respect. However no one has replied so far so it seems that there might be an issue here. Could you please check it out and guide me?
Regards,
Pankaj

**Comment by lilasquared** [comment]
@quasar-pankaj This is due to the typecasting and your inherited types.  in your sample the line including
```
var col = db.GetCollection<Person>("Employees").IncludeAll();
```
has nothing to include because `Person` has no collections direct collections and thus no dbrefs.  It is not until it is cast to `Employee` that the collections are there and able to be included.  If you change this line to 
```
var col = db.GetCollection<Employee>("Employees").IncludeAll();
```
the project names will no longer be null.  

That being said, there might be an underlying issue with the LiteCollection when it comes to type hierarchy like this.  My recommendation would be to not use DbRefs OR query the projects separately when the type of the returned person is an Employee.

**Comment by quasar-pankaj** [comment]
Thanks.

**Comment by lbnascimento** [comment]
Hi! With the objective of organizing our issues, we are closing old unsolved issues. Please check the latest version of LiteDB and open a new issue if your problem/question/suggestion still applies. Thanks!
